litellm/integrations/mlflow.py

```diff
@@ -0,0 +1,241 @@
+import json
+import threading
+from typing import Optional
+from litellm.integrations.custom_logger import CustomLogger
+from litellm.utils import verbose_logger
+
+import mlflow
+from mlflow.entities import Span, SpanEvent, SpanType, SpanStatusCode
+from mlflow.tracking import MlflowClient
+
+
+class MlflowLogger(CustomLogger):
+    def __init__(self):
+        self._client = MlflowClient()
+
+        self._stream_id_to_span = {}
+        self._lock = threading.Lock()  # lock for _stream_id_to_span
+
+
+    def log_success_event(self, kwargs, response_obj, start_time, end_time):
+        """
+        Log the success event as an MLflow span.
+        Note that this method is called asynchronously in the background thread.
+        """
+        try:
+            verbose_logger.debug(f"MLflow logging start for success event")
+
+            if kwargs.get("stream"):
+                self._handle_stream_event(kwargs, response_obj, start_time, end_time)
+            else:
+                span = self._start_span_or_trace(kwargs, start_time)
+                end_time_ns = int(end_time.timestamp() * 1e9)
+                self._end_span_or_trace(
+                    span=span,
+                    outputs=response_obj,
+                    status=SpanStatusCode.OK,
+                    end_time_ns=end_time_ns,
+                )
+        except Exception:
+            verbose_logger.debug(f"MLflow Logging Error", stack_info=True)
+
+
+    def log_failure_event(self, kwargs, response_obj, start_time, end_time):
+        """
+        Log the failure event as an MLflow span.
+        Note that this method is called *synchronously* unlike the success handler.
+        """
+        try:
+            verbose_logger.debug(f"MLflow logging start for failure event")
+
+            span = self._start_span_or_trace(kwargs, start_time)
+
+            end_time_ns = int(end_time.timestamp() * 1e9)
+
+            # Record exception info as event
+            exception_attributes = {
+                "exception.message": kwargs.get("exception"),
+                "exception.type": None,
+                "exception.stacktrace": kwargs.get("traceback_exception"),
+            }
+            event = SpanEvent(
+                name="exception",
+                timestamp=end_time_ns,
+                attributes=exception_attributes
+            )
+            span.add_event(event)
+
+            self._end_span_or_trace(
```

Comment:
is mlflow opentelemetry compatible? this looks pretty similar to OTEL, and i wonder if we can just leverage that integration here